central/auth/service/service_impl.go

```diff
@@ -9,10 +9,21 @@ import (
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/pkg/auth/user"
 	"github.com/stackrox/rox/pkg/grpc/authn"
-	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/or"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	authzUser "github.com/stackrox/rox/pkg/grpc/authz/user"
 	"google.golang.org/grpc"
 )
 
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		or.ServiceOr(authzUser.Authenticated()): {
+			"/v1.AuthService/GetAuthStatus",
+		},
```

Comment:
It seems that in the handler we duplicate most of the work we'll do in the authz middleware. What do you think about saying `allow.Anonymous()` here? Technically, a 403 or a 401 error is a somewhat valid response to `GetAuthStatus()` : ).

Reply:
Technically, the authorizers here do indeed duplicate a significant part of the work done later in the endpoint logic.
In the specific case of the `GetAuthStatus` function, returning `401 Unauthorised` to an anonymous (not authenticated) request is indeed valid.
I find it good when the authorizer documents the permissions required to use an endpoint. `allow.Anonymous()` here could be misleading as the service will require a valid identity to be present in the context to return a meaningful non-error response.
I'll switch the authorizer back to `allow.Anonymous()` with a comment explaining the choice of authorizer.